nft_ingester/src/bin/slot_persister/main.rs

```diff
@@ -497,3 +559,261 @@ async fn process_slots(
+                            }
+                        }
+                        next_slot = slots_iter.next();
+                        db_iter.next();
+                        current_db_slot = if db_iter.valid() {
+                            if let Some(key_bytes) = db_iter.key() {
+                                RawBlock::decode_key(key_bytes.to_vec()).ok()
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        };
+                    },
+                    Ordering::Less => {
+                        // Slot is missing in RocksDB
+                        {
+                            let mut missing_slots_lock = missing_slots.lock().unwrap();
+                            missing_slots_lock.push(slot);
+                        }
+                        // Advance slots iterator
+                        next_slot = slots_iter.next();
+                    },
+                    Ordering::Greater => {
+                        // slot > db_slot
+                        // Advance RocksDB iterator
+                        db_iter.next();
+                        current_db_slot = if db_iter.valid() {
+                            if let Some(key_bytes) = db_iter.key() {
+                                RawBlock::decode_key(key_bytes.to_vec()).ok()
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        };
+                    },
+                }
+            } else {
+                // No more slots in RocksDB, remaining slots are missing
+                {
+                    let mut missing_slots_lock = missing_slots.lock().unwrap();
+                    missing_slots_lock.push(slot);
+                    missing_slots_lock.extend(slots_iter);
+                }
+                break;
+            }
+        }
+
+        // Print missing slots
+        let missing_slots = Arc::into_inner(missing_slots).unwrap().into_inner().unwrap();
+        Some((lower_slot_boundary, last_valid_key, missing_slots))
+    }
+
+    let in_mem_dumper = Arc::new(InMemorySlotsDumper::new());
+    let slots_collector =
+        SlotsCollector::new(in_mem_dumper.clone(), backfill_source.clone(), metrics.clone());
+    let slot_backfill_offset = args.slot_backfill_offset;
+    loop {
+        let db = target_db.clone();
+        let dumper = in_mem_dumper.clone();
+        let collector = slots_collector.clone();
+        let (last_checkpoint, next_checkpoint, missed_slots) = get_missed_slots(
+            dumper,
+            collector,
+            db,
+            slot_backfill_offset,
+            cancellation_token.child_token(),
+        )
+        .await
+        .expect("retrieval of missed slots must join");
```

Comment:
is this a way to terminate the loop on receiving the cancelation token? As none is only returned on cancellation token termination